### ISSUE TITLE
Simplesearch result page showing assets from other pages

### ISSUE THREAD

**Issue by aradianoff** [getgrav/grav-plugin-simplesearch#9]
Whenever I make a search (doesn't matter which one), I find that in the results page ALL of the assets I load through 'shortcodes' plugin (haven't tried with assets plugin) in other pages are loaded in the head. 

Example here: http://aradianoff.com/es/search/query:grav
If you look in the `<head></head>`you can see scripts that shouldn't be there at all like the `<style></style>`and:
```
<link href="http://aradianoff.com/proyectos/bergondoabierto/cuentas-bergondo/sankey.css" type="text/css" rel="stylesheet" />
<script src="http://aradianoff.com/proyectos/bergondoabierto/cuentas-bergondo/sankey.js" type="text/javascript" defer></script>
<script src="http://aradianoff.com/proyectos/bergondoabierto/cuentas-bergondo/flows.js" type="text/javascript" defer></script>
<script src="http://aradianoff.com/proyectos/bergondoabierto/cuentas-bergondo/visualization.js" type="text/javascript" defer></script>
<script src="http://aradianoff.com/proyectos/bergondoabierto/cuantos-somos/vis.standford.edu.d3.min.js" type="text/javascript" defer></script>
<script src="http://aradianoff.com/proyectos/bergondoabierto/cuantos-somos/censo_2003_2014_edad_anual.js" type="text/javascript" defer></script>
```
That are all of the scripts I load with 'shortcodes' in other pages. This only seems to happen in the results page of simplesearch.

**Comment by Sommerregen** [comment]
Maybe this is a `Shortcodes` issue? But before you open another issue, I assume you get the same results when using the Assets plugin (since both are using `$grav['assets']->addCSS` etc. internally). In the past I had the same issues the other way round (CSS and JS was missing), but I couldn't resolved them... sorry!

Thus, I'm as curious as you waiting for @rhukster to answer :smile:

**Comment by rhukster** [comment]
Sorry to take so long to get back around to look at this.

So my initial thought is that this is not a bug, but actually an expected behavior.  If you were using a shortcode in the content, then you could expect that to render appropriately even in the summary of search results.  If these assets were not loaded, those shortcodes would result in incorrectly rendered output or just strange unexpected behavior. 

I'm going to close this, but if i've missed the mark (and it's quite possible!) reopen in.

**Comment by Sommerregen** [comment]
@rhukster @aradianoff This issue has been solved already in **Shortcodes v1.2.0**, too. No need to reopen it here. Assets through shortcodes is now being cached along with the page contents. Even if the cache is disabled, assets are only injected, when the respective page content is rendered now.
